### DISCUSSION TITLE
Tor Bridges,No Infura.io connections ,Custom RPC menu.

### DISCUSSION THREAD

**Discussion by OneNewSmartIdea** [horizontalsystems/unstoppable-wallet-android#5718] [Ideas]
Thank you for your awesome work. Your app has a great potential. Feels ages above MetaMask especially with the TOR option and the Github seems you really care about your work. 
Please add:
-Tor Bridges Menu
-Custom RPC menu (so we can add PulseChain blockchain the very second its live to do moves)
-Settings to unload parts of the features we don t use from the app so it doesn t feel like a casino or a CEX.

Please ditch any connection to Infura.io (My Firewall can resolves the IPs and I can see connections to Infura.io which doesn t mix with your own philosophy )

If you can do these changes I ll expose your project to some big communities that are angry of  MetaMask and Infura.io doing data collection. (MetaMask still connects to Infura.io even after you change the RPC to something else)
Thanks.

**Comment by esengulov** [comment]
thanks for the input, we are going to address following points in the near future, planned for release 0.31:

- No Infura.io connections
- Custom RPC menu
- Settings to unload parts of the features

As for Tor Bridges Menu we will digg into it as well but guessing won't be able to do in 0.31.

**Reply by esengulov** [reply]
> Settings to unload parts of the features
https://github.com/horizontalsystems/unstoppable-wallet-android/issues/5721

> Custom RPC menu
https://github.com/horizontalsystems/unstoppable-wallet-android/issues/5723

> No Infura.io connections
https://github.com/horizontalsystems/unstoppable-wallet-android/issues/5545

**Reply by OneNewSmartIdea** [reply]
I have tested Unstoppable wallet through Orbot together with a Chromium based browser and it works very nice. Rather than trying to implement further the Tor code it would be easier and more efficient to just integrate routing through Orbot. You can get some inspiration from the work of RethinkDNS Firewall app that already done this.

https://docs.rethinkdns.com/firewall/orbot (read the second method) 
https://github.com/celzero/rethink-app

**Comment by OneNewSmartIdea** [comment]
Wow, that was a fast answer. 
One more idea is that for the TOR mode it might be important to have a browser work through Tor also or even better use directly the original Tor browser app and use it as a proxy and this way you fix the bridges option and also give users the safety of their original code and not a custom implementation that would not be at the same maturation level. 
Thanks a lot guys. We will closely watch your progress.

**Comment by OneNewSmartIdea** [comment]
Btw. I m a GrapheneOS user, and I can say that they are the best private and secure mobile OS and your project seems to be the only one wallet app that can get on the same list for disruptive technologies in the space. Great work!

**Reply by esengulov** [reply]
hey @OneNewSmartIdea 

Following on our progress here. Basically, we are adding everything we have talked about earlier with the exception of the ability to add custom RPC network. We spent a few days on it and it turns out that at this point the feature would take us about two weeks due to some extensive refactoring. Therefore we decided to postpone it a bit. We are keen to implement it but a bit later on once we are done with some immediate backlog.

**Comment by OneNewSmartIdea** [comment]
I've tested the version 0.30 and I see there is an option now to add new RPC settings for the already existing networks, which is great progress and I look forward to see the option to add new Networks as I'm waiting for PulseChain to launch and I want to be able to use it as soon as it's live and also use their current test network.
BTW any info if you plan to support the BIP32 extended addresses generation? The same way MetaMask is doing it. This way many MetaMask users would be able to transfer their addresses. I already posted another message in the past in the Issues category about this and the answer was that it's doable but I didn't got an indication if it will be done or not. Thanks a lot for the great work you've done.
